packages/lean_analytics/lib/core/route_observer.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:flutter/material.dart';
+import 'package:lean_analytics/core/core.dart';
+import 'package:provider/provider.dart';
+
+abstract class LeanRoute extends MaterialPageRoute<void> {
+  LeanRoute(
+    this.name, {
+    required super.builder,
+    required super.settings,
+  });
+
+  final String name;
+}
+
+class GoRouterNavigationObserver extends NavigatorObserver {
+  GoRouterNavigationObserver(this.context);
+
+  final BuildContext context;
+
+  @override
+  void didPush(Route<dynamic> route, Route<dynamic>? previousRoute) {
+    if (route case LeanRoute()) {
+      context.read<LeanAnalytics>().register(
+            ScreenViewAnalyticsEvent(
+              pageName: route.name,
+              screenViewType: ScreenViewType.push,
+            ),
+          );
+    }
+  }
+
+  @override
+  void didPop(Route<dynamic> route, Route<dynamic>? previousRoute) {
+    if (previousRoute != null && previousRoute is LeanRoute) {
```

Comment:
You sometimes do if-case and sometimes normal checks

Reply:
unified this to case